src/installer/src/main.rs

```diff
@@ -0,0 +1,271 @@
+            Ok(app_prop) => app_prop
+                .lines()
+                .filter_map(|x| x.strip_prefix("application.version="))
+                .next()
+                .unwrap_or("?")
+                .to_string(),
+
+            Err(_) => "?".to_string(),
+        };
+
+        println!("Use Ghidra at: {} [v{}]? ({})", loc.display(), version, i);
+    }
+    println!("Abort ({})", good_ghidra_locations.len());
+
+    get_user_choice(good_ghidra_locations)
+}
+
+/// Determines Ghidra versions and provides selection interface for the user.
+fn get_user_choice(ghidra_locations: Vec<&PathBuf>) -> Result<PathBuf> {
+    println!("Please select (0-{}): ", ghidra_locations.len());
+
+    let mut choice = String::new();
+
+    std::io::stdin().read_line(&mut choice)?;
+
+    match choice.trim().parse::<usize>() {
+        Ok(i) if i == ghidra_locations.len() => Err(anyhow!("Installation canceled by user")),
+        Ok(i) if i < ghidra_locations.len() => Ok(ghidra_locations[i].clone()),
+        Ok(_) => Err(anyhow!("Invalid user input")),
+        Err(error) => Err(error.into()),
+    }
+}
+
+/// Creates ghidra.json for a Ghidra location at provided locaton.
+fn create_ghidra_json(location: &Path, ghidra_location: PathBuf) -> Result<()> {
+    let conf = GhidraConfig {
+        ghidra_path: ghidra_location,
+    };
+    println!("creating ghidra.json at: {}", location.display());
+    std::fs::create_dir_all(location)?;
+    std::fs::write(location.join("ghidra.json"), serde_json::to_string(&conf)?)?;
+    Ok(())
+}
+
+/// Runs Cargo install to install cwe_checker.
+fn install_cwe_checker() -> Result<()> {
+    std::process::Command::new("cargo")
+        .args(["install", "--path", "src/caller", "--locked"])
+        .status()?;
+    Ok(())
+}
+
+/// Recursive copy of files and directories.
+fn copy_dir_all(src: impl AsRef<Path>, dst: impl AsRef<Path>) -> Result<()> {
+    fs::create_dir_all(&dst)?;
+    for entry in fs::read_dir(src)? {
+        let entry = entry?;
+        let ty = entry.file_type()?;
+        if ty.is_dir() {
+            copy_dir_all(entry.path(), dst.as_ref().join(entry.file_name()))?;
+        } else {
+            fs::copy(entry.path(), dst.as_ref().join(entry.file_name()))?;
+        }
+    }
+    Ok(())
+}
+
+/// Copy src/ghidra to provided location.
+fn copy_ghidra_plugin(target: &Path) -> Result<()> {
+    let target = &target;
+    let mut source = env::current_dir()?;
+    source.push("src/ghidra");
+
+    copy_dir_all(source, target.join("ghidra"))?;
+    Ok(())
+}
+
+/// Removes provided locations and uninstalls cwe_checker via cargo
+fn uninstall(conf_dir: &Path, data_dir: &Path) -> Result<()> {
+    std::fs::remove_dir_all(conf_dir)?;
+    std::fs::remove_dir_all(data_dir)?;
+    std::process::Command::new("cargo")
+        .args(["uninstall", "cwe_checker"])
+        .status()?;
+    Ok(())
+}
+
+fn main() -> Result<()> {
+    let cwe_checker_conf_dir = ProjectDirs::from("", "", "cwe_checker").unwrap();
```

Comment:
Just to nitpick a little: `cwe_checker_conf_dir` contains both the config dir and the data dir for the cwe_checker, so the name of the variable is slightly misleading.